cockatrice/src/SpoilerBackgroundUpdaterThread.cpp

```diff
@@ -0,0 +1,233 @@
+#include <QDateTime>
+#include <QDebug>
+#include <QUrl>
+#include <QNetworkReply>
+#include <QMessageBox>
+#include <QFile>
+#include <QApplication>
+#include <QDesktopWidget>
+#include <QtConcurrent>
+#include "SpoilerBackgroundUpdaterThread.h"
+#include "settingscache.h"
+#include "carddatabase.h"
+#include "main.h"
+
+SpoilerBackgroundUpdaterThread::SpoilerBackgroundUpdaterThread(QObject *apParent) : QObject(apParent)
+{
+    // Determine if we're in spoilers season or not. If we're not, just kill this thread
+    downloadSpoilerSeasonStatusFile();
+    if (! mbIsActiveSpoilerSeason)
+    {
+        qDebug() << "Spoiler Timer disabled. NOT Spoilers Season";
+        return;
+    }
+
+    // Determine if the thread should be active in downloading/updating spoilers
+    mbIsActiveThread = settingsCache->getDownloadSpoilersStatus();
+    connect(settingsCache, SIGNAL(downloadSpoilerStatusChanged()), this, SLOT(changeActiveStatus()));
+
+    mpTimerForSpoilers = new QTimer(this);
+
+    // When timer ends, cause download of Spoilers XML file
+    connect(mpTimerForSpoilers, SIGNAL(timeout()), this, SLOT(timeoutOccurredTimeToDownloadSpoilers()));
+
+    // If there is a change in the settings, update the timer so we know when to update correctly
+    connect(settingsCache, SIGNAL(downloadSpoilerTimeIndexChanged()), this, SLOT(handleNewTimeInterval()));
+
+    // Start the timer sequence and the domino effects
+    runTimer();
+}
+
+SpoilerBackgroundUpdaterThread::~SpoilerBackgroundUpdaterThread()
+{
+    delete mpTimerForSpoilers;
```

Comment:
you should take care of deleting `mpNetworkAccessManager ` here, too, if it has been instanciated (!= nullptr).